### DISCUSSION TITLE
%ROWTYPE support?

### DISCUSSION THREAD

**Discussion by jcanada-gl** [sijms/go-ora#443] [Q&A] [answered]
I have a case where I have to call a function that has an "out" parameter of type [ROWTYPE](https://docs.oracle.com/cd/B13789_01/appdev.101/b10807/13_elems042.htm). Is there any way to get that type of data? Is there any chance that RegisterType works? I am currently using sqlx, and I think it is not compatible.

**Comment by sijms** [comment]
I think I should test and see what oracle return for this datatype then update the code
would you send an example code to test it

**Reply by jcanada-gl** [reply]
Yes, give me some time.

**Comment by jcanada-gl** [comment]
Ok, so if you take [THIS](https://github.com/sijms/go-ora/blob/master/examples/stored_proc/main.go) example, and replace the sqlText from de function _createStoredProc_ for this one: 
`CREATE OR REPLACE PROCEDURE GOORA_TEMP_GET_VISIT
(
	LVISIT_ID IN NUMBER,
	OUT_VISIT OUT GOORA_TEMP_VISIT%ROWTYPE
) AS
BEGIN
	SELECT * INTO OUT_VISIT
	FROM GOORA_TEMP_VISIT 
	WHERE VISIT_ID = LVISIT_ID;
END GOORA_TEMP_GET_VISIT;` 
should do the trick at oracle level. 


Let me know if this is what you where expecting.

How can you see what oracle it's returning?

**Comment by sijms** [comment, marked as answer]
sorry for late replay
I think procedure like this should be used inside `pl/sql` statement `(BEGIN... END)` because you should pass valid type for out_visit when calling the procedure/function otherwise oracle will return error

**Comment by sijms** [comment]
another solution is to create `custom type (user define type)` object with same fields as table and pass it to the function
